### ISSUE TITLE
Support Dex Web TLS options in Helm Chart

### ISSUE THREAD

**Issue by nabadger** [mintel/dex-k8s-authenticator#22]
Example here: https://github.com/coreos/dex/blob/master/examples/k8s/dex.yaml

Should end up with a Dex config like:
~~~
  https: 0.0.0.0:443
  tlsCert: /etc/dex/ssl/dex.crt
  tlsKey: /etc/dex/ssl/dex.key
~~~

`tlsCert` and `tlsKey` can be mounted by k8s volumes.

Also pay attention to service ports (http vs https). We're currently binding to 5556 (we should probably use port 80 or 443 by default).

**Comment by nabadger** [comment]
@whereisaaron - do you think it's reasonable to update the helm charts to default to port 80 for http (and 443 for https). At the moment it's the helm charts configure the service to port 5556 (or 5555 for dex-k8s-auth).

**Comment by whereisaaron** [comment]
Sure, it doesn't really matter or clash whatever ports the Pod container and Service use, and they each have their own cluster IP. And only want port that is exposed via an Ingress or NodePort matters to the end user. I don't think there is a convention or best practice on whether Service ports should match the front-end/external port or match the back end port.

If the `dex` default/native port is 5555 (which is seems to be), I'd suggest leaving at least the container running on that port, even if the Service then maps that to port 80 or 443 on the Service IP address.

**Comment by nabadger** [comment]
Started work on the [dex-helm-tls-support](/mintel/dex-k8s-authenticator/tree/dex-helm-tls-support) branch

TODO:
- [ ] Dex Supports listening on both https and http - config should support this
- [ ] DexK8sAuthenticator should support listening on both http and https (same as Dex)
- [ ] Replace `web` port name by `http` and `https`
- [ ] Current branch removes the ability to set path to .crt/.key files (may not be ok)
- [ ] There's a couple of typo's in the comments and a removed comment (re-add them)
- [ ] Add error handling in DexK8sAuthenticator for TLS options

Good reference points: 
* https://github.com/coreos/dex/blob/master/cmd/dex/config.go
* https://github.com/coreos/dex/blob/master/cmd/dex/serve.go

**Comment by nabadger** [comment]
This is in progress - I'll be pushing updates soon to the branch (hopefully by end of week).

**Comment by nabadger** [comment]
Will continue work on this after the UI updates are released.

**Comment by nabadger** [comment]
I've merged recent changes from #50. I'll close this for now and create a releases one documentation is updated as per #48
